### PR TITLE
[Owners Review] ViInt16[] support and added metadata in both Fgen and Fake

### PR DESCRIPTION
**Pull request by ni-pkhare**
### Justification

This PR incorporates mako changes related to handling ViInt16[] input array support. Along with it the APIs using it have also been added as part of metadata. 
Note: Unit Test for ViInt16[] input behaviour is also incorporated in the same PR.

### Implementation

ViInt16[] is treated as `repeated sint32` field. So First a vector<ViInt16> is created and then transform function is used to copy input array from gRPC request to the local vector<ViInt16> with the unary_operation of `[](auto x) { return (ViInt16)x; })`. Then later this local vector is passed to the C-API.

### Testing
Manually Tested.

### CODE REVIEW

File: generated/nifake/nifake_service.cpp [hunk 1]
```diff
@@ -911,7 +911,28 @@ namespace nifake_grpc {
...
+      ViInt32 input_array_sizes = request->input_array_of_integers().size();
+      auto input_array_of_floats = const_cast<ViReal64*>(request->input_array_of_floats().data());
+      auto input_array_of_integers_request = request->input_array_of_integers();
+      std::vector<ViInt16> input_array_of_integers;
+      std::transform(
+        input_array_of_integers_request.begin(),
+        input_array_of_integers_request.end(),
+        std::back_inserter(input_array_of_integers),
+        [](auto x) { return (ViInt16)x; });
+
```
Comment (Akshata440) [inline generated/nifake/nifake_service.cpp]: Remove the extra line here

File: generated/nifake/nifake_service.cpp [hunk 2]
```diff
@@ -911,7 +911,28 @@ namespace nifake_grpc {
...
+      ViInt32 output_array_size = request->output_array_size();
+      ViInt32 input_array_sizes = request->input_array_of_integers().size();
+      auto input_array_of_floats = const_cast<ViReal64*>(request->input_array_of_floats().data());
+      auto input_array_of_integers_request = request->input_array_of_integers();
+      std::vector<ViInt16> input_array_of_integers;
+      std::transform(
+        input_array_of_integers_request.begin(),
+        input_array_of_integers_request.end(),
+        std::back_inserter(input_array_of_integers),
+        [](auto x) { return (ViInt16)x; });
```
Comment (Akshata440) [inline generated/nifake/nifake_service.cpp]: Is a lambda required here? You could try giving just the `(ViInt16)x` as the last parameter in the transform

File: source/codegen/templates/service_helpers.mako [hunk 3]
```diff
@@ -208,6 +208,14 @@ ${initialize_standard_input_param(function_name, parameter)}\
...
         std::back_inserter(${parameter_name}),
         [](auto x) { return x ? VI_TRUE : VI_FALSE; });
+% elif c_type == 'ViInt16[]':
+      auto ${parameter_name}_request = ${request_snippet};
+      std::vector<${c_type_underlying_type}> ${parameter_name};
+      std::transform(
+        ${parameter_name}_request.begin(),
+        ${parameter_name}_request.end(),
+        std::back_inserter(${parameter_name}),
+        [](auto x) { return (${c_type_underlying_type})x; });
```
Comment (Akshata440) [inline source/codegen/templates/service_helpers.mako]: If you add a backslash at the end of this line, you wont see the extra line in the cpp file.

**Review by Akshata440 — APPROVED** [review]

File: source/codegen/metadata/nifgen/functions.py [hunk 4]
```diff
@@ -2902,6 +2974,71 @@
         ],
         'returns':'ViStatus'
     },
+    'WriteNamedWaveformI16': {
+        'codegen_method': 'public',
```
Comment (Akshata440) [inline source/codegen/metadata/nifgen/functions.py]: I dont think we need this codegen method tag here. If it is not needed, please remove it

File: source/codegen/common_helpers.py [hunk 5]
```diff
@@ -47,7 +47,7 @@ def is_unsupported_size_mechanism(parameter):
 def is_unsupported_scalar_array(parameter):
```
Comment (doshirohan) [inline source/codegen/common_helpers.py:47]: All conditions here now can be combined into single return statement.
```
return is_array(parameter['type']) and is_enum(parameter)
```

**Review by ganeshnie — APPROVED** [review]

**Review by doshirohan — APPROVED** [review]

File: generated/nifake/nifake_service.cpp (hunk 1 above)
Comment (ni-pkhare) [reply generated/nifake/nifake_service.cpp]: Done.

File: generated/nifake/nifake_service.cpp (hunk 2 above)
Comment (ni-pkhare) [reply generated/nifake/nifake_service.cpp]: The last parameter of transform function expects a function, hence used lambda.

File: source/codegen/templates/service_helpers.mako (hunk 3 above)
Comment (ni-pkhare) [reply source/codegen/templates/service_helpers.mako]: Done.

File: source/codegen/metadata/nifgen/functions.py (hunk 4 above)
Comment (ni-pkhare) [reply source/codegen/metadata/nifgen/functions.py]: Removed from all functions which were added in this PR.

File: source/codegen/common_helpers.py (hunk 5 above)
Comment (ni-pkhare) [reply source/codegen/common_helpers.py:47]: Makes sense, Done.

File: source/tests/unit/ni_fake_service_tests.cpp [hunk 6]
```diff
@@ -1421,6 +1421,29 @@ TEST(NiFakeServiceTests, NiFakeService_GetAnIviDanceWithATwistArray_CallsGetAnIv
     EXPECT_EQ(response.actual_size(), expected_size);
 }
 
+TEST(NiFakeServiceTests, NiFakeService_AcceptViInt16Array_CallsAcceptViInt16Array)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  std::int16_t int16_array[] = {0, 1, -0x8000, 0x7FFF};
```
Comment (jonlugoatx) [inline source/tests/unit/ni_fake_service_tests.cpp:1430]: Use constants from https://www.cplusplus.com/reference/climits/

**Review by jonlugoatx — APPROVED** [review]

**Pull request merged**